File: cakk-api/src/main/java/com/cakk/api/mapper/CakeDesignCategoryMapper.java
package com.cakk.api.mapper;

import java.util.List;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import com.cakk.common.enums.CakeDesignCategory;
import com.cakk.domain.mysql.entity.cake.CakeCategory;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class CakeDesignCategoryMapper {

	public static List<CakeCategory> supplyCakeCategoryListBy(final List<CakeDesignCategory> cakeDesignCategories) {
		return cakeDesignCategories
			.stream()
			.map(CakeDesignCategoryMapper::supplyCakeCategoryBy)
			.toList();
	}

	public static CakeCategory supplyCakeCategoryBy(CakeDesignCategory cakeDesignCategory) {
		return CakeCategory.builder()
			.cakeDesignCategory(cakeDesignCategory)
			.build();
	}
}

File: cakk-admin/src/main/kotlin/com/cakk/admin/mapper/CakeDesignCategoryMapper.kt
package com.cakk.admin.mapper

import com.cakk.common.enums.CakeDesignCategory
import com.cakk.infrastructure.persistence.entity.cake.CakeCategory

fun supplyCakeCategoryListBy(cakeDesignCategories: List<CakeDesignCategory>): List<com.cakk.infrastructure.persistence.entity.cake.CakeCategory> {
	return cakeDesignCategories.map { supplyCakeCategoryBy(it) }.toList()
}

private fun supplyCakeCategoryBy(cakeDesignCategory: CakeDesignCategory?): com.cakk.infrastructure.persistence.entity.cake.CakeCategory {
	return com.cakk.infrastructure.persistence.entity.cake.CakeCategory.builder()
		.cakeDesignCategory(cakeDesignCategory)
		.build()
}
